### PR TITLE
Create integration test profile and move tests

### PR DESCRIPTION
**Pull request by mocenas**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -233,6 +233,11 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <version>${maven-surefire-plugin.version}</version>
+        <configuration>
+          <excludes>
+            <exclude>**IT*</exclude>
```
Comment (jponge) [inline pom.xml:238]: You might want to use JUnit5 instead: https://junit.org/junit5/docs/current/user-guide/#writing-tests-tagging-and-filtering

File: api/src/test/java/com/redhat/insights/ITReportControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+    InsightsReportController instance =
+        InsightsReportController.of(logger, config, report, () -> httpClient);
+
+    // when trying to generate report, it should throw an exception, if opted out
+    assertThrows(InsightsException.class, instance::generate);
+
+    // give controller some time to send a report, if it wants
+    // but no reports should be sent
+    Thread.sleep(TimeUnit.SECONDS.toSeconds(2));
```
Comment (jponge) [inline api/src/test/java/com/redhat/insights/ITReportControllerTest.java:43]: We're using [Awaitility](http://www.awaitility.org/) in some other tests, it's always better to use it than a hard-coded `Thread.sleep`

File: api/src/test/java/com/redhat/insights/ITReportControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    InsightsReport report = mock(InsightsReport.class);
+    InsightsConfiguration config = new DefaultConfiguration();
+
+    InsightsReportController instance =
+        InsightsReportController.of(logger, config, report, () -> httpClient);
+    instance.generate();
+
+    // give controller some time to send a report, if it wants
+    // but no reports should be sent
+    Thread.sleep(TimeUnit.SECONDS.toSeconds(2));
```
Comment (jponge) [inline api/src/test/java/com/redhat/insights/ITReportControllerTest.java:62]: Same here

**Review by jponge — CHANGES_REQUESTED** [review]
Other than my comments this is a good start 👍

File: pom.xml (hunk 1 above)
Comment (mocenas) [reply pom.xml]: Good point, I changed it.

File: api/src/test/java/com/redhat/insights/ITReportControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,288 @@
+/* Copyright (C) Red Hat 2023 */
+package com.redhat.insights;
```
[the remark this answers is not in the document]
Comment (mocenas) [reply api/src/test/java/com/redhat/insights/ITReportControllerTest.java]: done

File: api/src/test/java/com/redhat/insights/ITReportControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+import com.redhat.insights.logging.InsightsLogger;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+
+public class ITReportControllerTest {
```
[the remark this answers is not in the document]
Comment (mocenas) [reply api/src/test/java/com/redhat/insights/ITReportControllerTest.java]: agree, done.

File: api/src/test/java/com/redhat/insights/ITReportControllerTest.java (hunk 2 above)
Comment (mocenas) [reply api/src/test/java/com/redhat/insights/ITReportControllerTest.java]: This is tricky. I changed Thread.sleep to Awaitility where it made sense. Thing is, in these test scenarios test want to give controller time to send a report, but no report should actually be send (there is something other blocking sending it).
Se basically here we want to wait and check that nothing happened. I don't see a way how awaility can really help in that.
I can change the code to something like `await().atLeast(2 seconds).checkSomething` but I don't really see a benefit in doing so.

**Pull request merged**